presto-main/src/test/java/com/facebook/presto/catalogserver/TestCatalogServerResponse.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.catalogserver;
+
+import com.facebook.airlift.json.JsonModule;
+import com.facebook.presto.common.QualifiedObjectName;
+import com.facebook.presto.connector.informationSchema.InformationSchemaTableHandle;
+import com.facebook.presto.connector.informationSchema.InformationSchemaTransactionHandle;
+import com.facebook.presto.metadata.HandleJsonModule;
+import com.facebook.presto.metadata.ViewDefinition;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.ConnectorMaterializedViewDefinition;
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.TableHandle;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.transaction.TransactionId;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestCatalogServerResponse
+{
+    private TestingCatalogServerClient testingCatalogServerClient;
+    private ObjectMapper objectMapper;
+
+    @BeforeTest
+    public void setup()
+    {
+        this.testingCatalogServerClient = new TestingCatalogServerClient();
```

Comment:
This only tests the client. Is it possible to add an e2e test which tests the client and the CatalogServer implementation?